lang/ruby/lib/avro/schema.rb

```diff
@@ -29,6 +29,8 @@ class Schema
     NAMED_TYPES_SYM     = Set.new(NAMED_TYPES.map(&:to_sym))
     VALID_TYPES_SYM     = Set.new(VALID_TYPES.map(&:to_sym))
 
+    NAME_REGEX = /^[A-Za-z_][A-Za-z0-9_.]*$/
```

Comment:
Hello!  This lets through names like `valid.9badPackage..9badName`.

For reference, the python regex is here: https://github.com/apache/avro/blob/f8afef1f98b1f27deb9e73ef804ed74ba2d20bbb/lang/py/src/avro/schema.py#L183

And Java scans names without a regex: https://github.com/RyanSkraba/avro/blob/1c5d3ff4ff78551edd4bab9824642d6918d3565d/lang/java/avro/src/main/java/org/apache/avro/Schema.java#L1518

(It looks like both Java and python also let `9badPackage` part through, but not the `9badName`...)

Reply:
Main issue is that the Ruby Avro library seems to assume that the namespace can be part of the name and it teases them apart later on, which is why it allows periods. Not quite sure why. I can add a check that there aren't two periods in a row.